README.md

```diff
@@ -47,20 +47,32 @@ store.dispatch(Action.SET_VISIBILITY_FILTER('SHOW_COMPLETED'));
 #### Match expressions
 
 ```ts
-const todosReducer = (state: Todo[] = [], action: Action) => Action.match({
-  // handle cases as pure functions instead of switch statements
-  ADD_TODO: ({ id, text }) => [
-    ...state,
-    { id, text, completed: false }
-  ],
-  TOGGLE_TODO: ({ id }) => state.map(todo =>
-    todo.id === id
-      ? {...todo, completed: !todo.completed}
-      : todo
-  )
-}, () => state /* default; if not provided, cases must be exhaustive */)(action);
+const todosReducer = (state: Todo[] = [], action: Action) =>
+  Action.match(action, {
+    // handle cases as pure functions instead of switch statements
+    ADD_TODO: ({ id, text }) => [...state, { id, text, completed: false }],
+    TOGGLE_TODO: ({ id }) =>
+      state.map(
+        todo =>
+          todo.id === id ? { ...todo, completed: !todo.completed } : todo
+      ),
+    // handles the rest; if not provided, cases must be exhaustive 
+    default: a => state // a===action. Useful for curried version. 
```

Comment:
It's a bit unfortunate now that the first example of `match` would be better implemented with `transform`.

Reply:
👍 I don't know what to do about it honestly :) 
Also when I tried to put docs for transform I kept questioning myself is it enough value to justify  extra api surface. I still like 70% positive on that.

Also take a look at my take on union https://github.com/twop/ts-union. `if` api is kinda somewhere in between match and transform. I'm wondering if unionize should have it instead

Reply:
I still think `transform` is and useful shorthand.

From a cursory inspection, it doesn't seem like `if` provides any more expressive power:

```ts
// typeof str === string
const str = PaymentMethod.if.Check(
  cash,
  n => `check num: ${n.toString()}`,
  _v => 'not check' // _v is the union obj that is passed in
); // str === 'not check'
```

vs

```ts
// typeof str === string
const str = PaymentMethod.match(
  cash,
  Check: n => `check num: ${n.toString()}`,
  default: _v => 'not check' // _v is the union obj that is passed in
); // str === 'not check'
```

Am I missing something?